main.go

```diff
@@ -378,12 +384,21 @@ func run(args []string) error {
 	}
 	metrics := sqmetrics.NewMetrics(*metricsURL, *metricsPrefix, client, *metricsInterval, metrics.DefaultRegistry, logger)
 
-	cert, err := buildCertificate(*keystorePath, *certPath, *keyPath, *keystorePass, *caBundlePath)
-	if err != nil {
-		logger.Printf("error: unable to load certificates: %s\n", err)
-		return err
+	var tlsConfigSource certloader.TLSConfigSource
+	if *useWorkloadAPI {
+		tlsConfigSource, err = certloader.TLSConfigSourceFromWorkloadAPI(*workloadAPIAddr, logger)
+		if err != nil {
+			logger.Printf("error: unable to create workload API TLS source: %s\n", err)
+			return err
+		}
+	} else {
+		cert, err := buildCertificate(*keystorePath, *certPath, *keyPath, *keystorePass, *caBundlePath)
+		if err != nil {
+			logger.Printf("error: unable to load certificates: %s\n", err)
+			return err
+		}
+		tlsConfigSource = certloader.TLSConfigSourceFromCertificate(cert)
```

Comment:
Might be nice to factor this out into a function. I guess in general I should probably refactor main() a bit to make it more readable.

Reply:
👍